src/canopy/knowledge_base/record_encoder/openai.py

```diff
@@ -42,9 +44,57 @@ def encode_documents(self, documents: List[KBDocChunk]) -> List[KBEncodedDocChun
         """  # noqa: E501
         return super().encode_documents(documents)
 
-    async def _aencode_documents_batch(self,
-                                       documents: List[KBDocChunk]
-                                       ) -> List[KBEncodedDocChunk]:
+    async def _aencode_documents_batch(
+        self, documents: List[KBDocChunk]
+    ) -> List[KBEncodedDocChunk]:
+        raise NotImplementedError
+
+    async def _aencode_queries_batch(self, queries: List[Query]) -> List[KBQuery]:
+        raise NotImplementedError
+
+
```

Comment:
Last but not least - please note that this class requires its own unit test before it can be merged.